cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/ClusterBootstrap.scala

```diff
@@ -108,8 +109,19 @@ final class ClusterBootstrap(implicit system: ExtendedActorSystem) extends Exten
     _selfContactPointUri.success(baseUri)
 
   /** INTERNAL API */
-  @InternalApi private[akka] def selfContactPoint: Future[Uri] =
-    _selfContactPointUri.future
+  @InternalApi private[akka] def selfContactPoints: Future[immutable.Seq[(String, Int)]] =
+    _selfContactPointUri.future.map { uri =>
+      if (system.settings.config.hasPath("akka.discovery.kubernetes-api.pod-domain") && uri.authority.host.isIPv4()) {
+        val podNamespace = system.settings.config.getString("akka.discovery.kubernetes-api.pod-namespace")
+        val podDomain = system.settings.config.getString("akka.discovery.kubernetes-api.pod-domain")
+        immutable.Seq(
+          (uri.authority.host.toString, uri.authority.port),
+          (s"${uri.authority.host.toString.replace('.', '-')}.${podNamespace}.pod.${podDomain}", uri.authority.port)
```

Comment:
which of those is used when we sort the addresses, or doesn't that matter?

Reply:
This doesn't matter: we sort the resolved targets, not the 'self contact point' (now 'self contact points')